src/zope/globalrequest/tests.py

```diff
@@ -1,37 +1,93 @@
-from unittest import TestSuite
-from zope.configuration import config
-from zope.configuration import xmlconfig
-from zope.testing.cleanup import cleanUp
-import doctest
-import zope.app.wsgi.testlayer
-import zope.globalrequest
-import zope.testbrowser.wsgi
+# -*- coding: utf-8 -*-
+import unittest
 
 
-def zcml(source):
-    context = config.ConfigurationMachine()
-    xmlconfig.registerCommonDirectives(context)
-    xmlconfig.string(source, context)
+class TestGlobalrequest(unittest.TestCase):
 
+    def tearDown(self):
+        # reset the threading local context
+        from threading import local
+        import zope.globalrequest.local
```

Comment:
So far I knew a couple of reasons to do this:

- A long time ago test module import errors were difficult to see -- they'd scroll by at the top of the test runner output, and be followed by the actual test run ending in "All 42 tests passed" or similar.  But test runners got fixed to report test module import errors properly, and we stopped importing inside test methods.

- I suppose when you import everything in the test method that makes it easier to move the test to a different module when refactoring.

I'd like to hear more about those "strange and difficult to debug side effects", if you've time.  Sounds interesting!